src/Files.App/Filesystem/StorageHistory/StorageHistoryOperations.cs

```diff
@@ -38,13 +42,13 @@ public async Task<ReturnResult> Undo(IStorageHistory history)
 				case FileOperationType.CreateNew: // Opposite: Delete created items
 					if (!IsHistoryNull(history.Source))
 					{
-						return await helpers.DeleteItemsAsync(history.Source, false, true, false);
+						return await helpers.DeleteItemsAsync(history.Source, UserSettingsService.FoldersSettingsService.ShowConfirmDeleteDialog, true, false);
```

Comment:
When I said we should try without a separate setting, I meant that we should show the confirmation regardless of the setting since it's not immediately clear that it will delete items. If user feedback comes in that they still want to hide the prompt in this case, we can add a separate setting and let them choose.

Reply:
I understand. I will do so.

Reply:
Now fixed.